### DISCUSSION TITLE
Support for Knowledge Distillation

### DISCUSSION THREAD

**Discussion by thangld201** [open-mmlab/mmclassification#1187] [Ideas]
Hi, I'm looking for a way to distil from ViT to MobilenetV2, given the checkpoints trained using mmclassification. This would involve using 2 dataloaders & models at the same time with a distillation loss (e.g. KLDivergence). I was looking for a way to initialize dataloader and model separately, without using Runner (since this is not supported), such that model's forward outputs include logits given a batch drawn from dataloader, but haven't figured out how to do so. Does anyone have any ideas ? I'm new to mmclassification.

**Comment by thangld201** [comment]
@mzr1996 Can you help me ?

**Reply by tonysy** [reply]
You can refer [MMRazor](https://github.com/open-mmlab/mmrazor) for some distillation example

**Comment by mzr1996** [comment]
We are planning to introduce some simple knowledge methods to mmcls in the future. But now, you can try to use MMRazor, it's a professional model compression tool for OpenMMLab.
